src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx

```diff
@@ -0,0 +1,406 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+import {
+  EuiSearchBarProps,
+  EuiBasicTableColumn,
+  EuiButtonIcon,
+  EuiConfirmModal,
+  EuiInMemoryTable,
+  EuiPopover,
+  EuiContextMenu,
+  EuiButton,
+  EuiTableSelectionType,
+  EuiEmptyPrompt,
+  EuiContextMenuPanelDescriptor,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { WorkspacePermissionSetting } from './types';
+import { WorkspacePermissionItemType, permissionModeOptions, typeOptions } from './constants';
+import { getPermissionModeId } from './utils';
+import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+import { AddCollaboratorButton } from './add_collaborator_button';
+import { WorkspaceCollaboratorType } from '../../services/workspace_collaborator_types_service';
+import {
+  WORKSPACE_ACCESS_LEVEL_NAMES,
+  accessLevelNameToWorkspacePermissionModesMap,
+} from '../../constants';
+import { WorkspaceCollaboratorAccessLevel } from '../../types';
+
+export type PermissionSetting = Pick<WorkspacePermissionSetting, 'id'> &
+  Partial<WorkspacePermissionSetting>;
+
+interface Props {
+  permissionSettings: PermissionSetting[];
+  onChange?: (value: PermissionSetting[]) => void;
+  displayedCollaboratorTypes: WorkspaceCollaboratorType[];
+  handleSubmitPermissionSettings: (permissionSettings: WorkspacePermissionSetting[]) => void;
+}
+
+export const WorkspaceCollaboratorTable = ({
+  permissionSettings,
+  onChange,
+  displayedCollaboratorTypes,
+  handleSubmitPermissionSettings,
+}: Props) => {
+  const [selection, setSelection] = useState<PermissionSetting[]>([]);
+  const { overlays } = useOpenSearchDashboards();
+
+  const handlePermissionSettingsChange = useCallback(
+    (newSettings) => {
+      onChange?.([...newSettings]);
+    },
+    [onChange]
+  );
+  const items = useMemo(() => {
+    return permissionSettings.map((setting) => {
+      const basicSettings = {
+        ...setting,
+        // This is used for table display and search match.
+        displayedType:
```

Comment:
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8486/files#diff-974bcd16c33ef28e4be36bfb140026ab9cb437f8b85bdd717ba7ecf58d38bd0eR17

Should we use this method to get the displayedType? I do not think it is `collaborator_table`'s duty to generate the displayedType.

Reply:
Yes, but this method not been implemented. It only provides a virtual type. cc @wanglam

Reply:
I think we should call all `typeOptions.getDisplayedType` and use the first not `undefined` result and display a `&mdash;` if it's undefined. Forget to register default `getDisplayedType` in my previous PR. We can call `getDisplayedType` here first.